### PR TITLE
Add support for all scalar types to `bsonproto`

### PR DESCRIPTION
**Pull request by AlekSi**

### CODE REVIEW

File: bsonproto/objectid.go [hunk 1]
```diff
@@ -1,3 +1,33 @@
 package bsonproto
 
-type ObjectID struct{}
+import "fmt"
+
+// ObjectID represents BSON scalar type ObjectID.
+type ObjectID [12]byte
+
+// SizeObjectID is a size of the encoding of [ObjectID] in bytes - 12.
```
Comment (cristaloleg) [inline bsonproto/objectid.go:8]: Also, do we need to document constant value at all? feels like a tautology.

File: bsonproto/objectid.go [hunk 2]
```diff
@@ -1,3 +1,33 @@
...
+const SizeObjectID = 12
+
+// EncodeObjectID encodes [ObjectID] value v into b.
+//
+// b must be at least 12 bytes long; otherwise, EncodeObjectID will panic.
+// Only b[0:12] bytes are modified.
+func EncodeObjectID(b []byte, v ObjectID) {
+	_ = b[11]
```
Comment (cristaloleg) [inline bsonproto/objectid.go:16]: Looks like this is the only place where we do such assert. Not a big deal but should we add in other places? or change next line to `copy(b[:12], v[:])`, dunno.

Comment (AlekSi) [reply bsonproto/objectid.go:16]: Functions like `binary.LittleEndian.Uint64` do that under the hood, too

Comment (cristaloleg) [reply bsonproto/objectid.go:16]: I know but do we need that?

Comment (cristaloleg) [reply bsonproto/objectid.go:16]: LOL, anyway.

File: bsonproto/objectid.go (hunk 1 above)
Comment (AlekSi) [reply bsonproto/objectid.go]: I think that all top-level symbols should be documented (or at least all _exported_ top-level symbols). But I'm all for using synonyms and other ways of removing tautology. Any suggestions?

File: bsonproto/time.go [hunk 3]
```diff
@@ -1 +1,33 @@
...
+	binary.LittleEndian.PutUint64(b, uint64(v.UnixMilli()))
+}
+
+// DecodeTime decodes [time.Time] value from b.
+//
+// If there is not enough bytes, DecodeTime will return a wrapped [ErrDecodeShortInput].
+func DecodeTime(b []byte) (time.Time, error) {
+	var res time.Time
+
+	if len(b) < 8 {
```
[the remark this answers is not in the document]
Comment (AlekSi) [reply bsonproto/time.go]: Done

File: bsonproto/time.go [hunk 4]
```diff
@@ -1 +1,33 @@
 package bsonproto
+
+import (
+	"encoding/binary"
+	"fmt"
+	"time"
+)
+
+// SizeTime is a size of the encoding of [time.Time] in bytes - 9.
```
[the remark this answers is not in the document]
Comment (AlekSi) [reply bsonproto/time.go]: Removed it

**Review by cristaloleg — APPROVED** [review]

File: bsonproto/objectid.go (hunk 1 above)
Comment (cristaloleg) [reply bsonproto/objectid.go]: I was trying to say that `// SizeObjectID is a size of the encoding of [ObjectID].` is more than enough :) 

Feel free to resolve.

Comment (AlekSi) [reply bsonproto/objectid.go]: Ah, then I did exactly that :)

**Pull request merged**